Formula/lilypond.rb

```diff
@@ -0,0 +1,98 @@
+class Lilypond < Formula
+  desc "Music notation for everyone"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "imagemagick" => :build
+  depends_on "pkg-config" => :build
+  depends_on "t1utils" => :build
+  depends_on "texi2html" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "guile@2"
+  depends_on :macos
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+  uses_from_macos "m4" => :build
+
+  resource "bison" do
+    url "https://ftp.gnu.org/gnu/bison/bison-3.7.6.tar.xz"
+    sha256 "67d68ce1e22192050525643fc0a7a22297576682bef6a5c51446903f5aeef3cf"
+  end
+
+  resource "urw-fonts" do
+    url "https://github.com/jsfelix/lilypond-build-deps/releases/download/v2.22.1/urw-base35-fonts-20200910.tar.gz"
+    sha256 "e0d9b7f11885fdfdc4987f06b2aa0565ad2a4af52b22e5ebf79e1a98abd0ae2f"
+  end
+
+  resource "texinfo" do
+    url "https://ftp.gnu.org/gnu/texinfo/texinfo-6.8.tar.xz"
+    sha256 "8eb753ed28bca21f8f56c1a180362aed789229bd62fff58bf8368e9beb59fec4"
+  end
```

Comment:
We should use the `texinfo` formula.

Reply:
If I add texinfo formula:
`Dependency 'texinfo' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.`

I need texinfo 6.1 or higher, and the one provided in macOS is older.
Is there a different way to do this?

Reply:
```ruby
depends_on "texinfo"
```

Reply:
It does not work in CI:
```
lilypond:
  * Dependency 'bison' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
  * Dependency 'texinfo' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
```

Reply:
And `uses_from_macs` crashes compiling

Reply:
This occours when I add `uses_from_macos`:
```
ERROR: Please install required programs:  bison >= 2.4.1 (installed: 2.3) makeinfo >= 6.1 (installed: 4.8)
```

Reply:
> It does not work in CI:
> 
> ```
> lilypond:
>   * Dependency 'bison' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
>   * Dependency 'texinfo' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
> ```

This is just a reminder to use the one that comes with macOS if we can. We can't, so we should do `depends_on "texinfo"` instead of building a throw-away `texinfo` in the formula.

Reply:
OK, but `brew audit --new lilypond` fails.

Reply:
ok, done. But i think the CI will fail because `brew audit --new lilypond` fails.